### PR TITLE
Feature: Refactor Build Configuration: Kotlin DSL + Version Catalog

### PR DESCRIPTION
**Pull request by wangerekaharun**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

- Migrated all Gradle build scripts from Groovy to Kotlin DSL.

- Introduced a centralized libs.versions.toml file for managing dependencies and plugin versions.

- Refactored dependency and plugin declarations to use the Gradle Version Catalog for cleaner and more consistent configuration.

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
- Migrated Gradle build scripts to Kotlin DSL

- Added centralized version catalog for dependencies

- Renamed example folder to sample

- Updated build configurations and dependencies

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td><strong>build.gradle.kts</strong><dd><code>Converted Groovy build script to Kotlin DSL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-29975cf604616657790eec0ca6230bffea93b3d887470f8c0e2097b61db8d479">+110/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>build.gradle.kts</strong><dd><code>Migrated sample app build script to Kotlin DSL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-4a2a6b9600679fa388d311ee670cbf2e19f1d6d90518f1864d77b53349a607c2">+46/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>build.gradle.kts</strong><dd><code>Added Kotlin DSL build configuration for sample</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-38c796f3418ddb0e0fad0484bd30222abcc35dc974feac0229d6500aae49407d">+21/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>gradle.properties</strong><dd><code>Enhanced Gradle performance configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-056ca4ce06cfd662376f62515a5cbfaf94fe3e9d0b1c4496e2b1a32c51f4f92e">+14/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.dart</strong><dd><code>Added sample Flutter app implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-0b03960429bdf3f24dba9e34a4b681eb355a6e9daef07cc2508460c0afb09113">[link]</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td><strong>libs.versions.toml</strong><dd><code>Added centralized version catalog for dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-baf4a872b28c79c64ddfd3741761d13df7f2942ae5bbf0a0f4a2ff2ab4775d4e">+27/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>settings.gradle.kts</strong><dd><code>Configured Gradle settings with version catalogs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-5916850af2ac36db5ff1e5374680c33c524ebbfa67fba76e18b8b157c0ec593b">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>build.yaml</strong><dd><code>Updated CI workflow for sample directory</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1">+7/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>release_ios.yml</strong><dd><code>Updated iOS release workflow for sample directory</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-0fecf8265161b43fb83f1b019bfd3dc589d29140e34d3e8125849cab2f332d0e">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>smile_id.podspec</strong><dd><code>Updated podspec with sample references</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-d2c2cb1c4f0a8f8b037553c17feb338491659f152b72dcb701a3ee9f0159a30d">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>widget_test.dart</strong><dd><code>Added basic Flutter widget test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-40e0f1f9dde34ecdb87fea9ebbfa13c000f90d0f21665eec2db886dc17e0cf49">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>plugin_integration_test.dart</strong><dd><code>Added integration test setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-4c8c9903eef15217799d3bceff672222bebe3904fa88fd80781222c15ad35206">[link]</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>CHANGELOG.md</strong><dd><code>Updated changelog with build system changes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>CONTRIBUTING.md</strong><dd><code>Updated contributing guide with sample references</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>SmileIDPluginTest.kt</strong><dd><code>Updated test path references</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-54c05a4401bee6337fbf265dd76a4d8dd7e47c9823a2d378a5390f30b718937c">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>gradle-wrapper.properties</strong><dd><code>Updated Gradle wrapper to newer version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-4c5ffadbdf6016477a1c3768e0b7381cc298bd558b1784871f407cbcb0efacce">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>42 files</summary><table>
<tr>
  <td><strong>settings.gradle.kts</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-b846529d3b9af5e34dfb0174742f9dc3d52bcd97761555bc61271a7a868870b9">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>gradle.properties</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-7b613b2cc241ffd66e44864c8113dffee067dd0685541821fb22e0270bf1bc66">+0/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-13d1bb8b37eeb4fabb1329de90134fd1276a4309ef42251a4f7b4695b9a56e0a">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>analysis_options.yaml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-65609710fe764b8312e73d8cab1ad5c519ce08af1bc76d40cd76cd38adb96b63">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AndroidManifest.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-c4a60ed1060420f672ddbae5c3ae10c3f93cc7ac90be45f4ce373b3df67cc126">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AndroidManifest.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-dc2ed174ecf9dfbd8f877c00bc31ba41ef63371972a4705a5a8dc8991c56e435">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>MainActivity.kt</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-a8d4787f0d091924b1306e0185c18770027c7cc7f9eb11787717a06cd8207dd9">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>launch_background.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-6a67b164e67ee00d8c09bc59ca4796ab049217e5df22a227db37cb3b5ac98578">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>launch_background.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-03e51146ba2de84f97deb1f59822b1184b296ca2ca1dd3279a87d30f90544e2e">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ic_launcher.webp</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-b67cf23adfd2130741d2c9eb579b49ea4d8e88bf069842756cad990f970443d1">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ic_launcher.webp</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-13bbd417291a882b89bb02a70b050eaf15f802d565322f81d64c69b9cf6b8f4e">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ic_launcher.webp</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-bd20ba27c7fb01a5a877e0f714bb128767186754345bb4eb7a46816c9b47e4bb">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ic_launcher.webp</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-365bd9cb940bd936a68f9c274464362bd53905e07769ec9ae5a1850f74676064">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ic_launcher.webp</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-d3af875ccf5a9f4b6a989ecaa7514f2f8cbeee180e0a96fdf60fb2016522ab18">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>styles.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-dc9afdf74ce2d4ffc71cecd38179c4c179cefaee1ced1014a99b77b4d52c5896">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>styles.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-87be946168bf3ad6ae51cd80d68125a327c5bd294df3e789e0549f0e05204daa">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AndroidManifest.xml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-ac93129d6b7f72f8140c3dc80eac0f607349df7d5bf592412c8e0f7ee5122e2a">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>gradle-wrapper.properties</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-564ec0ad322c7134a0c5a7c36f1ddb643836d8f7132f6713dff8e2129e21eded">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>gradlew</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-6535fc326b6d5efa49ead7d5fd8dae577b5257e8cc21026f45cc124446495455">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>gradlew.bat</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-83776da891218adf03c994c0f019bbdcbd7173968da991ee9a04877970b78ad6">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AppFrameworkInfo.plist</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-3c8d11a1e20e8f2d399a089621a92e9551746cffae3d689d31d7794ea806e66f">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Debug.xcconfig</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-0ae3ffa52fb0eee48f8719a7d2712c08264bc699b909bcd4333455320fb1d55c">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Release.xcconfig</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-01a1a3b446305ada3dd1020bd476ae9962afd92feb9ffd19eb74c85d3246e91f">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Podfile</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-464aef92ce3e1fcaca21a1f6f8284b189ffad648710c68f95de9abdf603bf886">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>project.pbxproj</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-bda5e6d5e7eed283fa6fd87af3eea62a81d50f2f8b56ec218405219693611b50">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>contents.xcworkspacedata</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-e986ea9bd057f2c45d5b90785df681310b57b7a8682e3596cfa4513fd8526e9f">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>IDEWorkspaceChecks.plist</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-66b243175c209ea9c7358ea40f8f07a158234616a7b8084118c243b32869561d">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>WorkspaceSettings.xcsettings</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-0db34a4306f3f82f8aa23035e22ee31e8fb45bf034cdd4aca6c8ae6e35277b8f">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Runner.xcscheme</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-f48465319b7814c513d4c421c4923ded788293b41d5f10fe7326286cda43e977">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>contents.xcworkspacedata</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-6d3039cd8361303eb9f72ff3b08e80cd1d3c1eb1d03b94a6acd0ffb0295b89ba">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>IDEWorkspaceChecks.plist</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-e9c7e7fbffa28d8713acc9e1c53270d8ff0ddfef9366415a3d97e9e1d10bac5d">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>WorkspaceSettings.xcsettings</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-62ab093c3bb2c8776155bfdf1f1c3e0c1e87f9a5485aed24bbb636de79c1dac3">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>AppDelegate.swift</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-3f216572cda4260134cabb45cc1906b68568da1c4faedad4b8048cb93774f4bf">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Contents.json</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-122cd5dca6e9b760900e18eba58b40c36c1277a7a78dbe2d561909b29c9d8e96">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Contents.json</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-2e648c03bfc5cb95f3a7f72791c7252fae866d877b9bf186a4c0da0e108dbac0">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-43259f49e399c8da4822516b18e09ce12145beac46f4faa8f91d19c88fd1198b">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LaunchScreen.storyboard</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-27823e7f36e9e0998814d5133e31181febff66a8c2c3bf47c31793599fab8c25">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Main.storyboard</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-311e34785d2b483db3c67ab97a3a839657942706fcdddc88756d8c1a88b8d183">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Info.plist</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-c0725f8d3516c853ee0680dcd8b3cd630ec43a802d5fbbfe90d9618dc76789ce">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Runner-Bridging-Header.h</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-b3265275a2c037afb8995fab0443565503d5f7356d4f1edf08da8d6faacb2f34">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>RunnerTests.swift</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-b28026c0e7a989c47437f51d3c74fcc3737dab9649e3129f8ab215e86c7679f4">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pubspec.yaml</strong></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/172/files#diff-4f70008358d8c5b76398b982681118e3dcc16bdc0a16ecb1e0f0d5ae20065215">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: android/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersionBuildScript")
+        }
+    }
+}
+
+allprojects {
+    repositories {
+        google()
+        mavenCentral()
+//      uncomment for development to test snapshots
```
Comment (jumaallan) [inline android/build.gradle.kts:39]: we can uncomment the snapshot maven url so we don't have to uncomment everytime we need to test a snapshot release

File: example/android/app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+plugins {
+    id("com.android.application")
```
Comment (jumaallan) [inline example/android/app/build.gradle.kts:2]: move to `libs.versions.toml` too 

all the plugins

File: android/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    sourceSets {
+        getByName("main").java.srcDirs("src/main/kotlin")
+        getByName("test").java.srcDirs("src/test/kotlin")
+    }
+
+    lint {
+        disable.add("NullSafeMutableLiveData")
+    }
+
```
Comment (jumaallan) [inline android/build.gradle.kts:99]: Can we also add this as a step here?

https://github.com/smileidentity/android/blob/844e755f7de29944b1c109aef29d46074f23ad40/sample/sample.gradle.kts#L95

We want to fail the build process if the `smile_config.json` is missing. The flutter app crashes with filePath not initialized when the config is missing and it's misleading

Comment (jumaallan) [reply android/build.gradle.kts:99]: You can also create a PR on android to move it to lib and not the sample app 

The build needs to fail in the lib (but not sure how that will happen since the asset folder is inside the sample app but we can discuss that later)

File: android/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        testLogging {
+            events("passed", "skipped", "failed", "standardOut", "standardError")
+            outputs.upToDateWhen { false }
+            showStandardStreams = true
+        }
+    }
+
+
+    buildFeatures.compose = true
+    if (!kotlinVersion.startsWith("2")) {
```
Comment (jumaallan) [inline android/build.gradle.kts:112]: I think we might need to create a plugin here to replace this check

It looks very basic

File: android/build.gradle.kts (hunk 1 above)
Comment (wangerekaharun) [reply android/build.gradle.kts:39]: Sure, will also move this to the `settings.gradle.kts` file since we're specifying the repositories there

File: example/android/app/build.gradle.kts (hunk 2 above)
Comment (wangerekaharun) [reply example/android/app/build.gradle.kts:2]: Yes still wip but the plan is to move all of them

File: android/build.gradle.kts (hunk 4 above)
Comment (wangerekaharun) [reply android/build.gradle.kts:112]: Yes I'm looking for best approaches to handle this

File: android/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+plugins {
+    alias(libs.plugins.android.library)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.ktlint)
+}
+
+// Apply compose plugin conditionally for Kotlin 2.x
+if (kotlinVersion.startsWith("2")) {
+    apply(plugin = "org.jetbrains.kotlin.plugin.compose")
```
Comment (jumaallan) [inline android/build.gradle.kts:53]: move this to `libs.versions.toml` too

File: android/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    buildFeatures.compose = true
+    if (!kotlinVersion.startsWith("2")) {
+        composeOptions {
+            kotlinCompilerExtensionVersion = kotlinCompilerExtension
+        }
+    }
+}
+
+dependencies {
+    implementation("com.smileidentity:android-sdk:$smileVersion")
```
Comment (jumaallan) [inline android/build.gradle.kts:119]: forgot to move these to `libs.versions.toml` too

File: example/android/build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+//}
+
+plugins {
+    alias(libs.plugins.flutter.plugin.loader)
+    alias(libs.plugins.android.application) apply false
+    alias(libs.plugins.kotlin.android) apply false
+    alias(libs.plugins.ktlint) apply false
+}
+
+//allprojects {
```
Comment (jumaallan) [inline example/android/build.gradle.kts:14]: if not used, delete then

File: sample/test/widget_test.dart [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // This is a basic Flutter widget test.
 //
 // To perform an interaction with a widget in your test, use the WidgetTester
-// utility in the flutter_test package. For example, you can send tap and scroll
+// utility in the flutter_test package. For sample, you can send tap and scroll
 // gestures. You can also use WidgetTester to find child widgets in the widget
```
Comment (jumaallan) [inline sample/test/widget_test.dart:5]: Undo this to example

**Review by jumaallan — APPROVED** [review]
